core/templates/services/literally-canvas-helper.service.spec.ts

```diff
@@ -0,0 +1,296 @@
+    LiterallyCanvasHelperService = $injector.get(
+      'LiterallyCanvasHelperService');
+  }));
+
+  it('should convert a rectangle shapeobject to svg tag', function() {
+    var rectShape = {
+      x: 142.5,
+      y: 96.5,
+      width: 12,
+      height: 29,
+      strokeWidth: 1,
+      strokeColor: 'hsla(0, 0%, 0%, 1)',
+      fillColor: 'hsla(0, 0%, 100%, 1)',
+      id: 'de569866-9c11-b553-f5b7-4194e2380d9f'
+    };
+    var actualSvgTag = '<rect id="rectangle-de569866-9c11-b553-f5b7-4194e23' +
+    '80d9f" x="143" y="97" width="12" height="29" stroke="hsla(0, 0%, 0%, 1' +
+    ')" fill="hsla(0, 0%, 100%, 1)" stroke-width="1"></rect>';
+    var svgTag = null;
+    svgTag = (
+      LiterallyCanvasHelperService.rectangleSVGRenderer(rectShape)
+    );
+    expect(svgTag).toBe(actualSvgTag);
+  });
+
+  it('should convert a ellpise shapeobject to svg tag', function() {
+    var ellipseShape = {
+      x: 60.5,
+      y: 77.5,
+      width: 30,
+      height: 45,
+      strokeWidth: 2,
+      strokeColor: 'hsla(0, 0%, 0%, 1)',
+      fillColor: 'hsla(0, 0%, 100%, 1)',
+      id: '4343fcbf-b1e9-3c6d-fcc8-809c00c6ba9b'
+    };
+    var renderedSvgTag = '<ellipse id="ellipse-4343fcbf-b1e9-3c6d-fcc8-809c0' +
+    '0c6ba9b" cx="75.5" cy="99.5" rx="15" ry="22" stroke="hsla(0, 0%, 0%, 1)' +
+    '" fill="hsla(0, 0%, 100%, 1)" stroke-width="2"></ellipse>';
+    var svgTag = null;
+    svgTag = (
+      LiterallyCanvasHelperService.ellipseSVGRenderer(ellipseShape)
+    );
+    expect(svgTag).toBe(renderedSvgTag);
+  });
+
+  it('should convert a text shapeobject to svg tag', function() {
+    var textShape1 = {
+      x: 72.5,
+      y: 142.5,
+      text: 'hello',
+      color: 'hsla(0, 0%, 0%, 1)',
+      font: '18px \"Helvetica Neue\",Helvetica,Arial,sans-serif',
+      v: 1,
+      id: '90ee8761-dd62-9d70-b61a-02e6fec487e9'
+    };
+    var textShape2 = {
+      x: 72.5,
+      y: 142.5,
+      text: 'hello',
+      renderer: {
+        lines: ['hello']
+      },
+      color: 'hsla(0, 0%, 0%, 1)',
+      font: '18px \"Helvetica Neue\",Helvetica,Arial,sans-serif',
+      v: 1,
+      id: '90ee8761-dd62-b61a-9d70-02e6fec487e9'
+    };
+    var renderedSvgTag1 = '<text id="text-90ee8761-dd62-9d70-b61a-02e6fec487' +
+    'e9" x="72.5" y="142.5" fill="hsla(0, 0%, 0%, 1)" style="font: 18px &quo' +
+    't;Helvetica Neue&quot;, Helvetica, Arial, sans-serif;"><tspan x="72.5" ' +
+    'dy="0" alignment-baseline="text-before-edge">hello</tspan></text>';
+    var renderedSvgTag2 = '<text id="text-90ee8761-dd62-b61a-9d70-02e6fec487' +
+    'e9" x="72.5" y="142.5" fill="hsla(0, 0%, 0%, 1)"' +
+    ' style="font: 18px &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-s' +
+    'erif;"><tspan x="72.5" dy="0" alignment-baseline="text-before-edge">hel' +
+    'lo</tspan></text>';
+    var svgTag = null;
+    svgTag = (
+      LiterallyCanvasHelperService.textSVGRenderer(textShape1)
+    );
+    expect(svgTag).toBe(renderedSvgTag1);
+    svgTag = (
```

Comment:
Create another variable here. Using just one variable and reassigning it can make it hard to understand

Reply:
Done.